mslice.py

```diff
@@ -9,12 +9,14 @@ class MsliceGui(QMainWindow,Ui_MainWindow,MainView):
     def __init__(self):
         super(MsliceGui,self).__init__()
         self.setupUi(self)
-        workspace_presenter = self.wgtWorkspacemanager.get_presenter()
-        self._presenter = MainPresenter(self,workspace_presenter)
+        self._presenter = MainPresenter(self)
 
-        self.wgtWorkspacemanager.set_main_window(self)
-        self.wgtSlice.set_workspace_selector(self)
-        self.wgtPowder.set_workspace_selector(self)
+        self.wgtWorkspacemanager.get_presenter().register_master(self)
```

Comment:
This isn't quite right. It should be the presenters that are registered and not the view and `self` here refers to the view. 

How about passing the 3 presenters, manager, slice & powder, to the `MainPresenter` constructor and then having that call `register_master(self)`?